app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt

```diff
@@ -427,24 +429,62 @@ class SettingsFragment : PreferenceFragmentCompat() {
     }
 
     private fun getClickListenerForMakeDefaultBrowser(): Preference.OnPreferenceClickListener {
-        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-            Preference.OnPreferenceClickListener {
-                val intent = Intent(android.provider.Settings.ACTION_MANAGE_DEFAULT_APPS_SETTINGS)
-                startActivity(intent)
-                true
+        return when {
+            Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q -> {
```

Comment:
Can you add a description to this function explaining the scenarios for each version? Looks like we specify the listener for Q and N